base/server/python/pki/server/cli/acme.py

```diff
@@ -323,3 +333,203 @@ def execute(self, argv):
+        self.add_module(ACMEDatabaseModifyCLI())
+
+
+class ACMEDatabaseShowCLI(pki.cli.CLI):
+
+    def __init__(self):
+        super(ACMEDatabaseShowCLI, self).__init__(
+            'show', 'Show ACME database configuration')
+
+    def print_help(self):
+        print('Usage: pki-server acme-database-show [OPTIONS]')
+        print()
+        print('  -i, --instance <instance ID>       Instance ID (default: pki-tomcat).')
+        print('  -v, --verbose                      Run in verbose mode.')
+        print('      --debug                        Run in debug mode.')
+        print('      --help                         Show help message.')
+        print()
+
+    def execute(self, argv):
+
+        try:
+            opts, _ = getopt.gnu_getopt(argv, 'i:v', [
+                'instance=',
+                'verbose', 'debug', 'help'])
+
+        except getopt.GetoptError as e:
+            logger.error(e)
+            self.print_help()
+            sys.exit(1)
+
+        instance_name = 'pki-tomcat'
+
+        for o, a in opts:
+            if o in ('-i', '--instance'):
+                instance_name = a
+
+            elif o in ('-v', '--verbose'):
+                logging.getLogger().setLevel(logging.INFO)
+
+            elif o == '--debug':
+                logging.getLogger().setLevel(logging.DEBUG)
+
+            elif o == '--help':
+                self.print_help()
+                sys.exit()
+
+            else:
+                logger.error('Unknown option: %s', o)
+                self.print_help()
+                sys.exit(1)
+
+        instance = pki.server.instance.PKIServerFactory.create(instance_name)
+
+        if not instance.exists():
+            raise Exception('Invalid instance: %s' % instance_name)
+
+        instance.load()
+
+        acme_conf_dir = os.path.join(instance.conf_dir, 'acme')
+        database_conf = os.path.join(acme_conf_dir, 'database.conf')
+        config = {}
+
+        logger.info('Loading %s', database_conf)
+        pki.util.load_properties(database_conf, config)
+
+        database_class = config.get('class')
+
+        database_type = DATABASE_TYPES.get(database_class)
+        print('  Database Type: %s' % database_type)
+
+        if database_type == 'postgresql':
+
+            url = config.get('url')
+            if url:
+                print('  Server URL: %s' % url)
+
+            username = config.get('user')
+            if username:
+                print('  Username: %s' % username)
+
+            password = config.get('password')
+            if password:
+                print('  Password: ********')
+
+
+class ACMEDatabaseModifyCLI(pki.cli.CLI):
+
+    def __init__(self):
+        super(ACMEDatabaseModifyCLI, self).__init__(
+            'mod', 'Modify ACME database configuration')
+
+    def print_help(self):
+        print('Usage: pki-server acme-database-mod [OPTIONS]')
+        print()
+        print('  -i, --instance <instance ID>       Instance ID (default: pki-tomcat).')
```

Comment:
Do we want a `-k/--key` and `-u/--value` to enable non-interactive key/value changes? Then this can be utilized in automation &c more easily than an interactive CLI would.

Reply:
Yes, I'm planning to add something like that after this.

Reply:
\o/ cool -- glad we're on the same page, just jumping the gun as usual.

Reply:
No worries :) I did the interactive first since I'm trying to update the docs as well, but the non-interactive will follow soon.